### PR TITLE
feat(driver): track L2 execution engine's P2P sync progress

### PR DESCRIPTION
**Pull request by davidtaikocha**
- rename `L2 node` to `L2 execution engine`  ( some flag names were updated @alexshliu  )
- track L2 execution engine's P2P sync progress
- rename some variables

### CODE REVIEW

File: driver/sync_progress_tracker.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	syncProgessFetchInterval = 10 * time.Second
```
Comment (alexshliu) [inline driver/sync_progress_tracker.go:16]:
```suggestion
	syncProgressFetchInterval = 10 * time.Second
```

File: driver/sync_progress_tracker.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+	return s.triggered
+}
+
+// LastSyncedVerifiedBlockID returns tracker.lastSyncedVerifiedBlockID.
+func (s *BeaconSyncProgressTracker) LastSyncedVerifiedBlockID() *big.Int {
+	s.mutex.RLock()
+	defer s.mutex.RUnlock()
+
+	return s.lastSyncedVerifiedBlockID
```
Comment (johntaiko) [inline driver/sync_progress_tracker.go:194]: maybe return a copy instead of pointer to avoid data-race

Comment (davidtaikocha) [reply driver/sync_progress_tracker.go:194]: I think there wont be a data-race because of the `Rlock()` in line 191 ?

Comment (davidtaikocha) [reply driver/sync_progress_tracker.go]: changed: https://github.com/taikoxyz/taiko-client/pull/61/commits/12fb3cc0fc13cef0ccdbfe23d178f47fde2e73a4

**Review by johntaiko — APPROVED** [review]

**Review by alexshliu — APPROVED** [review]

**Pull request merged**